### PR TITLE
refactor: use smart eqeqeq null checks

### PR DESCRIPTION
**Pull request by Shinigami92**
Instead of explicitly checking on undefined, I refactored the code to relax-checking on null with a non-strict `==`
These are checking for "null or undefined" and called smart null checks.
The benefit of them is, that folks that are using plain JS without type-checking can also pass `null` as an `undefined`/optional parameter and get the same behavior as not passing a value.

### CODE REVIEW

**Comment by ST-DDT** [comment]
Is there a lint rule we can enable to enforce these checks?

**Review by ST-DDT — APPROVED** [review]

**Comment by ST-DDT** [comment]
It feels like the usage of `v == null`, `!v`, `v != null`, ... is inconsistent.

**Comment by Shinigami92** [comment]
> It feels like the usage of `v == null`, `!v`, `v != null`, ... is inconsistent.

It is not, cause `!0` is `false`. So there are some places where we explicitly count e.g. a length that is `0` as a `false` case and then we fall back to e.g. a defined value like `4`

**Comment by ST-DDT** [comment]
This needs a rebase

**Review by ST-DDT — DISMISSED** [review]

File: src/date.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ export class _Date {
     to: string | Date | number,
     num?: number
   ): Date[] {
-    if (typeof num === 'undefined') {
+    if (num == null) {
       num = 3;
     }
```
Comment (ST-DDT) [inline src/date.ts:128]: Just set it as default.

**Review by pkuczynski — APPROVED** [review]

**Pull request merged**